server/src/main/java/org/tctalent/server/repository/db/JobSpecification.java

```diff
@@ -153,6 +155,33 @@ public static Specification<SalesforceJobOpp> buildSearchQuery(
                 conjunction.getExpressions().add(published);
             }
 
+            //UNREAD MESSAGES
+            if (request.getWithUnreadMessages() != null && request.getWithUnreadMessages()) {
```

Comment:
Thanks for the suggestion. I didn't merge it in because although it is shorter, I don't think it is clearer. The way it is currently written is consistent with the processing with all other request fields: namely, if the request field is present, process it. In this case, if the Boolean field is present, use its value.